dev-docs/design/how-lit-works.md

```diff
@@ -0,0 +1,302 @@
+# How lit-html Works
+
+## About this Document
+
+This is an overview for how lit-html works, what makes it fast, and how the code is organized. It's intended for contributors, or anyone who is interested in the inner workings of lit-html.
+
+All the source code described by this document can be found in [`lit-html.ts`](https://github.com/lit/lit/blob/main/packages/lit-html/src/lit-html.ts).
+
+The source code itself is generally optimized to be read and we try to be as clear and self-documenting as possible, erring on the side of verbosity, with numerous comments when needed to explain why code is written as it is. We expect the source to be optimized by tools, so we can keep it understandable by humans.
+
+If anything is unclear or you have questions, reach out on our [Discord channel](https://discord.com/invite/buildWithLit), or ask a question on [GitHub Discussions](https://github.com/lit/lit/discussions). All our communities can be found on the [Lit Community page](https://lit.dev/docs/resources/community/).
+
+# What is lit-html?
+
+lit-html is an HTML templating library. Templates are written in JavaScript by mixing static HTML strings and dynamic JavaScript values using template literals. lit-html enables a functional / UI-as-data programming model, fast initial rendering, and fast updates that minimally update DOM when state changes.
+
+The key feature that enables this is separating static parts of templates from the dynamic parts with template literals, and never traversing or updating the static parts after the initial render. This makes lit-html very performant as can be seen in the [JS Frameworks Benchmark](https://krausest.github.io/js-framework-benchmark/). A great companion to this document is [Justin Fagnani's lit-html talk](https://youtu.be/Io6JjgckHbg).
+
+# Foundational Pieces
+
+lit-html relies on two key platform features for its performance and ergonomics: [JavaScript Tagged Template Literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates) and the HTML [`<template>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) element.
```

Comment:
The discussion of this primitives here kind of breaks the flow of the doc imo. I think I'd remove the sections on TTL's and template and just had a sentence here for each saying why they are used. Something like: TTLs are a convenient way to mix dynamic js values and static strings of html. The static strings are immutable and therefore can be a cache key for a <template> element that expresses this static html as DOM...

Reply:
Agree - done